src/communication/bluetooth/chunker/BluetoothChunker.cpp

```diff
@@ -0,0 +1,164 @@
+#include "communication/bluetooth/chunker/BluetoothChunker.h"
+
+BluetoothChunker::BluetoothChunker(BLEService* pService, const char* cuuid, const uint32_t access) {
+  characteristic = pService->createCharacteristic(cuuid, access);
+  BLECallback = new BLEChunkerCallback(this);
+  characteristic->setCallbacks(BLECallback);
+};
+
+BluetoothChunker::~BluetoothChunker() {
+  delete characteristic;
+  delete BLECallback;
+}
+
+bool BluetoothChunker::startReceiving(std::string& headerChunk) {
+  if(isActiveReceiving()) {
+    Logger::error("[BluetoothChunker::startReceiving()] Reveiving transaction already started");
+    return false;
+  }
+  
+  if(headerChunk.length() < headerSize) {
+    Logger::error("[BluetoothChunker::startReceiving()] Invalid message");
+    return false;
+  }
+
+  toReceiveChunkCount = 0;
+  //copy num save in many chars to single long type variable
+  for(int i = 0; i < headerSize; i++) {
+    // save sizeof(char) bytes
+    toReceiveChunkCount = toReceiveChunkCount | headerChunk[i]; 
+    // move all bytes of 8 byte in left, make space for next part of num
+    if(i < headerSize - 1) toReceiveChunkCount = toReceiveChunkCount << 8;
+  }
+  
+  //remove header
+  headerChunk = headerChunk.substr(headerSize);
```

Comment:
New variable named `contentChunk` will be better (and then we don't have to have separate comment to show whats going on)

Reply:
Necessary it's must name headerChunk because I modify first chunk whois passing to function by reference, but I can change the comment to more informative `//remove the header from original chunk`